src/utils/periods/relativePeriods.js

```diff
@@ -195,3 +240,118 @@ export const flattenedRelativePeriods = RELATIVE_PERIODS.reduce(
+    const firstMonthStr = (firstMonthIndex + 1).toString().padStart(2, '0')
+    const firstMonthStartDate = new Date(`${year}-${firstMonthStr}-01`)
+
+    return firstMonthStartDate.setMonth(
+        firstMonthStartDate.getMonth() - offset * unitSize
+    )
+}
+
+const yearsAgo = (offset, isFinancial) => {
+    const currentYear = getCurrentDate().getFullYear()
+    const startYear = currentYear - offset
+    return new Date(isFinancial ? `${startYear}-10-01` : `${startYear}-01-01`)
+}
+
+const startDateCalculators = {
+    [RELATIVE_PERIOD_CODES.thisDay]: () => daysAgo(0),
+    [RELATIVE_PERIOD_CODES.yesterday]: () => daysAgo(1),
+    [RELATIVE_PERIOD_CODES.last3Days]: () => daysAgo(3),
+    [RELATIVE_PERIOD_CODES.last7Days]: () => daysAgo(7),
+    [RELATIVE_PERIOD_CODES.last14Days]: () => daysAgo(14),
+    [RELATIVE_PERIOD_CODES.thisWeek]: () => weeksAgo(0),
+    [RELATIVE_PERIOD_CODES.lastWeek]: () => weeksAgo(1),
+    [RELATIVE_PERIOD_CODES.last4Weeks]: () => weeksAgo(4),
+    [RELATIVE_PERIOD_CODES.last12Weeks]: () => weeksAgo(12),
+    [RELATIVE_PERIOD_CODES.last52Weeks]: () => weeksAgo(52),
+    [RELATIVE_PERIOD_CODES.weeksThisYear]: () => weeksThisYear(),
+    [RELATIVE_PERIOD_CODES.thisMonth]: () => monthUnitsAgo(1, 0),
+    [RELATIVE_PERIOD_CODES.lastMonth]: () => monthUnitsAgo(1, 1),
+    [RELATIVE_PERIOD_CODES.last3Months]: () => monthUnitsAgo(1, 3),
+    [RELATIVE_PERIOD_CODES.last6Months]: () => monthUnitsAgo(1, 6),
+    [RELATIVE_PERIOD_CODES.last12Months]: () => monthUnitsAgo(1, 12),
+    [RELATIVE_PERIOD_CODES.monthsThisYear]: () => startOfYear(),
+    [RELATIVE_PERIOD_CODES.thisBimonth]: () => monthUnitsAgo(2, 0),
+    [RELATIVE_PERIOD_CODES.lastBimonth]: () => monthUnitsAgo(2, 1),
+    [RELATIVE_PERIOD_CODES.last6BiMonths]: () => monthUnitsAgo(2, 6),
+    [RELATIVE_PERIOD_CODES.biMonthsThisYear]: () => startOfYear(),
+    [RELATIVE_PERIOD_CODES.thisQuarter]: () => monthUnitsAgo(3, 0),
+    [RELATIVE_PERIOD_CODES.lastQuarter]: () => monthUnitsAgo(3, 1),
+    [RELATIVE_PERIOD_CODES.last4Quarters]: () => monthUnitsAgo(3, 4),
+    [RELATIVE_PERIOD_CODES.quartersThisYear]: () => startOfYear(),
+    [RELATIVE_PERIOD_CODES.thisSixMonth]: () => monthUnitsAgo(6, 0),
+    [RELATIVE_PERIOD_CODES.lastSixMonth]: () => monthUnitsAgo(6, 1),
+    [RELATIVE_PERIOD_CODES.last2SixMonths]: () => monthUnitsAgo(6, 2),
+    [RELATIVE_PERIOD_CODES.thisFinancialYear]: () => yearsAgo(0, true),
+    [RELATIVE_PERIOD_CODES.lastFinancialYear]: () => yearsAgo(1, true),
+    [RELATIVE_PERIOD_CODES.last5FinancialYears]: () => yearsAgo(5, true),
+    [RELATIVE_PERIOD_CODES.thisYear]: () => yearsAgo(0),
+    [RELATIVE_PERIOD_CODES.lastYear]: () => yearsAgo(1),
+    [RELATIVE_PERIOD_CODES.last5Years]: () => yearsAgo(5),
+}
+
+export const getRelativePeriodStartDate = periodCode => {
+    const startDateCalutator = startDateCalculators[periodCode]
+
+    if (!startDateCalutator) {
+        return null
+    }
+
+    return formatYyyyMmDd(startDateCalutator())
```

Comment:
Personally I think I'd go with a simpler name for this function. The pascalcase formatted date reads a bit awkward.

Reply:
This is similar to https://github.com/dhis2/reports-app/pull/192#discussion_r500906665. I'd like to sticvk with the naming applied in `@dhis2/analytics`.

Reply:
Sounds good 👍